### DISCUSSION TITLE
link components url sometimes wrong, and new tab never work

### DISCUSSION THREAD

**Discussion by samg666** [Budibase/budibase#3698] [Help]
Hello,

I use link components on a repeater, and bind URL something like `/orders/details/{{ RPT_Orders.FetchOrdersByStatus.OrderNum }}/{{RPT_Orders.FetchOrdersByStatus.RelServiceID}}/orders_summary`, then the app sometimes generated link like `http://192.168.100.14:10000/app_518e74d1b3c744e89da7d10f18f156fa#/orders/details/OD210390/CD1074-106/orders_summary` and it can be accessed normal. But sometimes it's something like `http://192.168.100.14:10000/orders/details/OD210390/CD1074-106/orders_summary`, then got `AccessDenided` error. Is there any of my setting wrong, or it's a bug?

![iShot2021-12-07 10 38 54](https://user-images.githubusercontent.com/29649651/144956115-967f35f6-3b64-466f-9bac-180da1538cc9.png)

BTW, the link `New Tab` seems never worked. Suppose if the `New Tab` was checked, the link should be open in a new tab/page, but it never worked for me.

![image](https://user-images.githubusercontent.com/29649651/144956867-2c3d5634-0f67-4945-a2d8-3d956c1b1294.png)

**Comment by mjashanks** [comment]
So if set the link to 
- `sales/{{ New Table.Sales._id }}` - I can reproduce your access denied issue
- `/sales/{{ New Table.Sales._id }}` (leading `/`) - It works ok

Also - your `New Tab` issue - it seems to be working only for external links - e.g. `https://google.com`. I will log this issue.

**Comment by samg666** [comment]
Hi @mjashanks , I am sure all my links are leading with `/`. But the generated link sometimes are different.

When the generated link like below with `app_518e74d1b3c744e89da7d10f18f156fa`, then it's normal.
`http://192.168.100.14:10000/app_518e74d1b3c744e89da7d10f18f156fa#/orders/details/OD210390/CD1074-106/orders_summary`

But in some case, the generated link like below without app id, then access denied. I am not sure how to reproduce it, but it dose happens. Many of my users complained this issue, the workaround is refresh the page or re-open the page.
`http://192.168.100.14:10000/orders/details/OD210390/CD1074-106/orders_summary`

BTW, is there any way to show custom page title on the browser, instead of show app name?

![image](https://user-images.githubusercontent.com/29649651/145054424-44dfca8c-db89-473b-be29-19f135b1a7d0.png)

**Comment by samg666** [comment]
Hi @mjashanks , I have finally find out the link always binded a wrong URL after doing a search (the data provider / repeater was refreshed).

When the page was 1st opened, the url is something like `http://192.168.100.14:10000/app_518e74d1b3c744e89da7d10f18f156fa#/demo/details/2`, with an app id `app_518e74d1b3c744e89da7d10f18f156fa#`, this one is correct.
![iShot2021-12-08 10 30 25](https://user-images.githubusercontent.com/29649651/145138226-c08d9530-2e90-4e21-ba5b-5221fb602f31.png)

But when I doing a search, the url become something like `http://192.168.100.14:10000/demo/details/106`, the app id `app_518e74d1b3c744e89da7d10f18f156fa#` was removed, then will got `AccessDenied` error. 
![iShot2021-12-08 10 31 44](https://user-images.githubusercontent.com/29649651/145138243-e1940db8-6d4d-4666-9f62-60a2e8a12e62.png)

**Comment by samg666** [comment]
Hi @mjashanks , I can reproduce this issue on the template apps, it must be a bug. I am running on version 0.9.173.

https://user-images.githubusercontent.com/29649651/145140543-a9b01893-a067-49b8-9a64-cbfb4f9f3878.mov

**Reply by mjashanks** [reply]
Thanks so much for tracking this down. I can reproduce this - we'll look into it asap.
